### DISCUSSION TITLE
keyboard shortcuts help

### DISCUSSION THREAD

**Discussion by aloispichler** [i3/i3#5234] [Ideas]
Would be important to have a keyboard shortcut helper, triggered by pressing $mod+F1, say.
Even Qtile has that now: https://github.com/qtile/qtile/issues/1329W

**Comment by ghost** [comment]
You can have all your `bindsym` only commands into a separate file & source it to original config.
Then have a shortcut like `bindsym $mod+F1 $TERMINAL -e $EDITOR <location/to/your/bindsym-config>`

**Comment by aloispichler** [comment]
Thank you, I like this idea!

The following hack reads the i3/config file, reports the bindings and what they do in a new window:
![Screenshot_2022-11-03_15-15-19](https://user-images.githubusercontent.com/45781387/199745225-7a5b42eb-383a-42eb-a31a-8560a3347fa3.png)
Every line in the i3 config file (`~/.config/i3/config`) starting with `'##'` and followed by `'bindsym'` in the next line is considered to be its description.

The shortcut _mod+F1_ displays the i3 keybindings:
```
##	Keybindings
bindsym $mod+F1 exec alacritty -e ~/.config/i3/configKeys.sh
```
The (executable) script `~/.config/i3/configKeys.sh` uses _fzf_, for convenience, and python:
```
#!/bin/bash
# ~/.config/i3/configKeys.sh
python ~/.config/i3/configKeys.py | fzf --prompt="Search for keybind: " --border=rounded --margin=1% --color=dark --height 100% --reverse --header="i3 CHEAT SHEET" --info=hidden --header-first
```
The python file `~/.config/i3/configKeys.py` reads the config file:
```
#   python ~/.config/i3/configKeys.py
#   reads /i3/config and reports 'bindsym' if they have a preceeding description indicated by '##'

import os
def showKeys():
    keyHelp = ''
    linePrev =  ''
    with open(os.path.expanduser(r".config/i3/config"), 'r') as fp:      # full path to the i3/config
        for line in fp:
            if 'bindsym' in line and len(linePrev)> 2 and linePrev[0:2] == '##':
                key = line.split()[1]                 # grab the keybinding
                desc= linePrev[2:].strip().title()    # grab its description
                if len(keyHelp) > 0:
                    keyHelp += "\n"
                keyHelp += "{:<17} {}".format(key, desc)
            linePrev= line
    return keyHelp

print(showKeys())
```

Any improvements are welcome.
